src/core/compiler.jl

```diff
@@ -288,47 +255,63 @@ function build_model_info(input_expr)
 end
 
 """
-    translate_tilde!(model_info)
+    replace_vi!(model_info)
 
-Translates ~ expressions to observation or assumption expressions, updating `model_info`.
+Replaces @vi() expressions to the VarInfo instance.
 """
-function translate_tilde!(model_info)
+function replace_vi!(model_info)
+    ex = model_info[:main_body]
+    vi = model_info[:main_body_names][:vi]
+    ex = MacroTools.postwalk(x -> @capture(x, @vi()) ? vi : x, ex)
```

Comment:
Is the `macro vi` defined somewhere or is it just used as a placeholder?

Also, should we use the `vi`, given that we already have a `vi` function used in variational inference, i.e. completely different purpose?

Reply:
The macro is not defined. I can define and export it just to give a warning if someone tries to define another macro with the same name. I can change the name of the macro to `varinfo`.

Reply:
> The macro is not defined. I can define and export it just to give a warning if someone tries to define another macro with the same name. 

Is there a reason why it looks like a macro then? 
And yeah, might be a good idea to define and export as you suggested:)

> I can change the name of the macro to varinfo. Does that sound better?

Sounds good to me!:)

Btw, awesome work on this PR! :heart:

Reply:
Thanks! I wanted to make it a macro because I didn't want to conflict with other variables. This use case is rare enough that we don't want to reserve a variable name for it. So I just felt that it is much harder to define a macro `@varinfo` by mistake than it is to define a variable. And if a macro with this name is defined in the user's module, when we have exported `@varinfo` from Turing, (unless in REPL I believe) a warning will be given to the user so there is less possibility of making a mistake and calling `@varinfo` thinking it is doing something when in fact we are just replacing it with our `VarInfo`.